### PR TITLE
Expose bean setters in @Configuration used by @EnableWebFluxSecurity

### PR DESCRIPTION
**Pull request by jgrandja**
Fixes gh-6624

### CODE REVIEW

File: config/src/main/java/org/springframework/security/config/annotation/web/reactive/WebFluxSecurityConfiguration.java [hunk 1]
```diff
@@ -49,12 +49,20 @@
...
-	@Autowired
 	ApplicationContext context;
 
+	@Autowired(required = false)
+	void securityWebFilterChains(List<SecurityWebFilterChain> securityWebFilterChains) {
+		this.securityWebFilterChains = securityWebFilterChains;
+	}
+
+	@Autowired
+	void context(ApplicationContext context) {
```
Comment (sdeleuze) [inline config/src/main/java/org/springframework/security/config/annotation/web/reactive/WebFluxSecurityConfiguration.java]: After having a look to the implementation, I am wondering if we could not avoid this `ApplicationContext` and just leverage regular bean autowiring which would be conceptually cleaner IMO.

Comment (jgrandja) [reply config/src/main/java/org/springframework/security/config/annotation/web/reactive/WebFluxSecurityConfiguration.java]: Good point - I reverted that change. I applied the same for  `ServerHttpSecurityConfiguration.beanFactory` as well so reverted that as well.

FYI, I did a force-push to start fresh.

**Comment by jgrandja** [comment]
I also feel that it can be confusing to a user looking at this code and wondering why is a `@Configuration` class is being used by a functional API in it's implementation.

I agree with @rwinch, it might be the right direction to take as far as

> If it needs the logic in `ServerHttpSecurityConfiguration`, perhaps we should extract out another class that can be used by both `@Configuration` and the functional code?

**Comment by sdeleuze** [comment]
@rwinch In Spring Fu case, the annotations are indeed not used, but I would like to leverage these classes to create the same infra via function bean registration like I do for Spring Boot, Spring MVC or WebFlux configuration. If the principle is not clear, please have a look to https://github.com/spring-projects/spring-fu/tree/master/autoconfigure-adapter. These initializers are leveraged by Kofu and Jafu DSLs. From my understanding, these configuration classes are used in a functional way and are needed to make it possible to have a `ServerHttpSecurity` instance. If my understanding is wrong, please let me know.

**Comment by jgrandja** [comment]
@sdeleuze I spoke with @rwinch and we came up with the following approach:

```java
ServerHttpSecurityConfiguration configuration = new Configurer()
		.adapterRegistry(adapterRegistry)
		.authenticationManager(authenticationManager)
		.reactiveUserDetailsService(reactiveUserDetailsService)
		.passwordEncoder(passwordEncoder)
		.userDetailsPasswordService(userDetailsPasswordService)
		.configuration();
```

Does this work for you?

**Comment by jgrandja** [comment]
@sdeleuze As per our discussion, you don't require a fluent configuration API so I've provided you standard setters with package-private access and moved `@Autowired` from field to these new setters.

**Comment by sdeleuze** [comment]
That's exactly what I need for programmatic use, and I like how it improves configuration testability, so :+1: from me.

**Pull request merged**